<?xml version="1.0"?>
<launch>

    <arg name="isVehicle" default="false"/>
    <arg name="isOperator" default="true"/>
    <arg name="InvertSteeringInGearReverse" default="false"/>
    <arg name="command_creation_param_path" default="$(find-pkg-share tod_command_creation)/config/tod_command_creation.param.yaml" />
    
    <group if="$(var isOperator)">
        <push-ros-namespace namespace="Operator/InputDevices"/>
        <node pkg="tod_command_creation" 
        exec="OperatorCommandCreator" 
        name="command_creator" 
        output="screen">

        <param from="$(var command_creation_param_path)" />   
        </node>

    </group>

    <group if="$(var isVehicle)">
        <push-ros-namespace namespace="Vehicle/CommandCreation"/>
        <node name="SecondaryCmdReceiver" pkg="tod_command_creation" exec="VehicleSecondaryCommandReceiver" output="screen"/>
        <node name="PrimaryCmdReceiver" pkg="tod_command_creation" exec="VehiclePrimaryCommandReceiver" output="screen"/>
    </group>

</launch>
